src/sentry/rules/history/preview.py

```diff
@@ -256,7 +256,11 @@ def get_top_groups(
     Since frequency conditions require snuba query(s), we need to limit the number groups we process.
     """
     if has_issue_state_condition:
-        datasets = {dataset_map.get(group) for group in condition_activity.keys()}
+        datasets = {
+            dataset_map.get(group)
+            for group in condition_activity.keys()
+            if dataset_map.get(group) is not None
```

Comment:
I suspect this can just be `dataset_map[group]` -- or at least I would hope so :shrug:

Reply:
I wasn't entirely sure, but agreed. I'll change it and fix this if it ends up being problematic.